pkg/sloop/ingress/kubewatcher_test.go

```diff
@@ -222,7 +224,11 @@ func Test_existingOrStartNewCrdInformer(t *testing.T) {
 	kw := &kubeWatcherImpl{protection: &sync.Mutex{}}
 	kw.crdInformers = make(map[crdGroupVersionResourceKind]*crdInformerInfo)
 
-	client := dynamicFake.NewSimpleDynamicClient(runtime.NewScheme())
+	gvrToListKind := map[schema.GroupVersionResource]string{
+		{Group: "g", Version: "v", Resource: "r"}: "kList",
+	}
+
+	client := dynamicFake.NewSimpleDynamicClientWithCustomListKinds(runtime.NewScheme(), gvrToListKind)
```

Comment:
I suppose the new client is backward compatible @karlskewes ?

Reply:
> kubectl is supported within one minor version (older or newer) of kube-apiserver. - https://kubernetes.io/releases/version-skew-policy/#kubectl

Where `kubectl` is using the client SDK like Sloop.
Since Kubernetes upstream changed from supporting 3 versions to 4 there is potential for a version skew greater than 1.
I haven't had too many issues with running larger skews in practice in other situations, except when API kind's are deprecated. 
For Sloop we need to solve PDB per #251.

In testing this PR it looked like the resources were discovered in both v1.22 and v1.26.
There as an existing issue #214 regarding updating support for new Kubernetes Event version. This PR doesn't solve that.

So there are a few things to do and I thought it would be best to tackle one by one, sharing solution as go.
We could build additional PR's off of this and when we have the full story merge them.
Or build on top, or take commits from here.

Then finally when all are solved we can tag a new release.

Reply:
Sounds good. Thanks @karlskewes